src/main/java/se/kth/sda/skeleton/comments/CommentController.java

```diff
@@ -1,8 +1,72 @@
 package se.kth.sda.skeleton.comments;
 
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import java.util.List;
 
+/**
+ * Represents the controller layer (or the API). This exposes application functionality of Comment as RESTfull webservices.
+ */
 @RestController
 public class CommentController {
+    private CommentService commentService;
 
+    /**
+     * Constructs a CommentController and automatically assigns its {@code commentService} field.
+     * @param commentService an object that implements interface CommentService
+     */
+    @Autowired
+    public CommentController(CommentService commentService){
+        this.commentService = commentService;
+    }
+
+    /**
+     * Handler method for endpoint "/posts/{id}/comments" with GET HttpRequest. Returns a {@link ResponseEntity}
+     * containing the list of Comments associated with a Post of the given {@code postId} and HTTP status {@code OK}.
+     * @param id the Post whose entire list of Comments is returned
+     * @return {@link ResponseEntity} containing the list of Comments associated with a Post of the given {@code postId}
+     * and HTTP status {@code OK}
+     */
+    @GetMapping("/posts/{id}/comments")
+    public ResponseEntity<List<Comment>> getAllComments(@PathVariable Long id){
+        return ResponseEntity.ok(commentService.getAllComments(id));
+    }
+
+    /**
+     * Handler method for endpoint "/posts/{id}/comments" with POST HttpRequest. Returns a {@link ResponseEntity} containing the created
+     * comment and HTTP status {@code OK}. Requires that the HttpRequest from client include Comment object {@code comment}
+     * which is the Comment to be created.
+     * @param id the Post for whom a new Comment is created
+     * @param comment the comment that will be associated with the Post whose id is {@code id}
+     * @return {@link ResponseEntity} containing the created Comment and HTTP status {@code OK}
+     */
+    @PostMapping("/posts/{id}/comments")
+    public ResponseEntity<Comment> createComment(@PathVariable Long id, @RequestBody Comment comment){
+        commentService.createComment(id, comment);
+        return ResponseEntity.status(HttpStatus.CREATED).body(comment);
+    }
+
+    /**
+     * Handler method for endpoint "/comments/{id}" with GET HttpRequest. Returns a {@link ResponseEntity} containing a Comment with the
+     * given {@code id} and HTTP status {@code OK}.
+     * @param id the id used to find the Comment associated with it
+     * @return {@link ResponseEntity} containing a Comment with the given {@code id} and HTTP status {@code OK}
+     */
```

Comment:
javadoc does not match the method. Doc relates to a GET comment by id method but the code is for a PUT method

Reply:
Add a Get comment by id method

Reply:
done